src/test_l1_dsVideoPort.c

```diff
@@ -92,9 +92,113 @@ static int gTestID = 1;
 	}\
 }\
 
+#define MAX_RESOLUTIONS 10
+typedef struct {
+    int type;
+    int index;
+} ConnectedAudioPort_t;
+
+typedef struct {
+    uint16_t port_no;
+    uint16_t type_id;
+    char* name;
+    bool dtcp_supported;
+    bool hdcp_supported;
+	uint16_t numSupportedResolutions;
+    char* supportedResolutionNames[MAX_RESOLUTIONS];
+    ConnectedAudioPort_t connected_audio_ports;
+    char* defaultResolution; // Adjust based on expected resolution name length
+    int colorspaces;
+    int supported_color_depth_capabilities;
+    bool display_surrond;
+    int supported_tv_resolutions_capabilities;
+    int hdr_capabilities;
+    int hdcp_protocol_version;
+    int quantization_ranges;
+    int matrix_coefficients;
+} PortConfig_t;
+
 /* Global flags to support features */
 static bool extendedEnumsSupported=false; //Default to not supported
-
+static char* deviceType = NULL;
+static char* deviceName = NULL;
+static int number_of_ports = 0;
+static char* port_types_str = NULL;
+static int color_depth = 0;
+PortConfig_t *ports = NULL;
+
+void populate_device_config() {
+   	ut_kvp_status_t status;
+	char key[256];
+   	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "Device/Type", deviceType, UT_KVP_MAX_ELEMENT_SIZE);
+   	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "Device/Name", deviceName, UT_KVP_MAX_ELEMENT_SIZE);
+	number_of_ports = ut_kvp_getUInt16Field(ut_kvp_profile_getInstance(), "Number_of_ports");
+   	status = ut_kvp_getStringField(ut_kvp_profile_getInstance(), "Port_types[0]", port_types_str, UT_KVP_MAX_ELEMENT_SIZE);
+	color_depth = ut_kvp_getUInt16Field(ut_kvp_profile_getInstance(), "color_depth");
+
+    ports = (PortConfig_t *)malloc(number_of_ports * sizeof(PortConfig_t));
+    if (ports == NULL) {
+        printf("Failed to allocate memory for ports array");
+		return;
+    }
+	for (int i = 0; i < number_of_ports; i++) {
+		ports[i].port_no = ut_kvp_getUInt16Field(ut_kvp_profile_getInstance(), "Ports/i/PortNo");
```

Comment:
need to replace Ports/i/PortNo with sscanf string

Reply:
Done